src/pages/api/signUp.ts

```diff
@@ -0,0 +1,29 @@
+import bcrypt from 'bcrypt'
+import connectDB from "@/lib/mongodb";
+import User from "@/types/User";
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+    if (req.method !== 'POST') {
+        return res.status(405).end()
+    }
+    const { name, email, password, rePassword } = req.body
+    if (password !== rePassword) {
+        return res.status(400).json({ error: 'Password miss match' })
+    }
+
+    await connectDB()
+    const isUserExist = await User.findOne({ email })
+    if (isUserExist) {
+        return res.status(409).json({ error: 'User alredy exist' })
+    }
+
+    try {
+        const salt = 10
+        const hashedPassword = await bcrypt.hash(password, salt)
+        const newUser = await User.create({ email, name, password: hashedPassword })
+        res.status(201).json({ message: 'User is sign up', user: newUser })
```

Comment:
Here the message is also not in a proper English. In my opinion just don't return the `message` property at all just the user. 201 already indicates that it is successful.